WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java

```diff
@@ -57,8 +72,42 @@ private void addIntervals(IntervalManager intervalManager) {
 		intervals.addAll(intervalManager.getOpenIntervals());
 	}
 
-	/** Filters out and removes intervals which are older than one hour. */
+	/**
+	 * Filters out and removes intervals which are older than 10 hours from
+	 * Database. Filters intervals for selected time span.
+	 */
 	private void filterIntervals() {
+		int numberOfMinutes = 600;
+
+		switch (selectedInterval) {
+		case MINUTES_10:
+			numberOfMinutes = 10;
+			break;
+
+		case MINUTES_30:
+			numberOfMinutes = 30;
+			break;
+
+		case HOUR_1:
+			numberOfMinutes = 60;
+			break;
+
+		case HOURS_2:
+			numberOfMinutes = 120;
+			break;
+
+		case HOURS_5:
+			numberOfMinutes = 300;
+			break;
+
+		case HOURS_8:
+			numberOfMinutes = 480;
+			break;
+
+		case HOURS_10:
+			numberOfMinutes = 600;
+		}
```

Comment:
Great suggestion, thanks! Just another field in enum StatisticsInterval makes everything more elegant.